gcache/src/gcache_page_store.cpp

```diff
@@ -76,9 +76,16 @@ remove_file (void* __restrict__ arg)
     pthread_exit(NULL);
 }
 
+/*
+ * Returns false if there are no more pages to be deleted (either
+ * the queue is empty or if the first page is in use).
+ * Otherwise, returns true.
+*/
 bool
 gcache::PageStore::delete_page ()
 {
+    if (pages_.empty()) return false;
```

Comment:
This can now happen because of (!keep_size_ && !keep_page_) condition in cleanup(). This condition was added to keep the behavior the same when nothing is set (using the default values of keep_size_=0 and keep_page_=0).